crates/agentgateway/src/telemetry/metrics.rs

```diff
@@ -120,6 +170,9 @@ pub struct Metrics {
 	pub tcp_downstream_tx_bytes: Family<TCPLabels, counter::Counter>,
 
 	pub upstream_connect_duration: Histogram<ConnectLabels>,
+
+	// metrics for guardrail trips (mask/reject) for request/response
+	pub guardrail_trips: Family<GuardrailLabels, counter::Counter>,
```

Comment:
'trips' seems a bit confusing for me. I struggled to understand if it was attempts or failures or roundtrips etc (at first, I get it now)

Maybe guardrail_checks and we can have allow/mask/reject instead of just mask/reject? knowing the success seems handy